TextSelection/Views/MultitouchTextView.swift

```diff
@@ -0,0 +1,225 @@
+        
+        let before = self.textStorage.string.nonWhitespaceCharacterIndexBefore(index)
+        let after = self.textStorage.string.nonWhitespaceCharacterIndexAfter(index)
+        
+        let newSelection = Selection(first: before, last: after, attributes: [NSForegroundColorAttributeName: highlightTextColor, NSBackgroundColorAttributeName: highlightSelectionColor])
+        
+        self.highlightSelection = newSelection.add(self.highlightSelection)
+        self.preSelection = nil
+        
+        self.updateViews(false)
+        self.selectionDidChange()
+    }
+    
+    func continueHighlighting(point: CGPoint) {
+        guard self.highlightSelection != nil else {
+            return
+        }
+        
+        // not sure I need this
+        // self.updateTextManagers()
+        
+        let index = self.layoutManager.glyphIndexForPoint(point, inTextContainer: self.textContainer, fractionOfDistanceThroughGlyph: nil)
+        
+        if (index > self.highlightSelection!.first) {
+            self.highlightSelection!.first = self.highlightSelection!.start
+            self.highlightSelection!.last = self.textStorage.string.nonWhitespaceCharacterIndexAfter(index)
+        } else {
+            self.highlightSelection!.first = self.highlightSelection!.end
+            self.highlightSelection!.last = self.textStorage.string.nonWhitespaceCharacterIndexBefore(index)
+        }
+        
+        self.updateViews(false)
+        self.selectionDidChange()
+    }
+    
+    func endHighlighting(point: CGPoint) {
+        guard self.highlightSelection != nil else {
+            return
+        }
+        
+        self.preSelection = self.highlightSelection!.add(self.preSelection)
+        self.preSelection?.attributes = [NSForegroundColorAttributeName: preSelectionTextColor, NSBackgroundColorAttributeName: preSelectionColor]
+        self.highlightSelection = nil
+        self.updateViews(false)
+        self.selectionDidChange()
+        
+        self.clearPreSelectionWithDelay()
+    }
+    
+    func appendPreSelectionToSelection() {
+        guard let preSelection = self.preSelection else {
+            return
+        }
+        self.selection = preSelection.add(self.selection)
+        self.selection?.attributes = [NSForegroundColorAttributeName: selectionTextColor, NSBackgroundColorAttributeName: selectionColor]
+        self.preSelection = nil
+        self.updateViews(false)
+        self.selectionDidChange()
+    }
+    
+    func clearSelection() {
+        self.preSelectionTimer?.invalidate()
+        self.highlightSelection = nil
+        self.preSelection = nil
+        self.selection = nil
+        self.updateViews(false)
+        self.selectionDidChange()
+    }
+    
+    func updateViews(animated: Bool) {
+        if (animated) {
+            UIView.transitionWithView(self.textInputView, duration: 0.3, options: [.TransitionCrossDissolve], animations: {
+                self.updateTextAttributes()
```

Comment:
Can't use the usual "animateWithDuration" here because it's not changing normal view frame-ish properties. We're changing the attributed string properties.